pkg/scale/decode_test.go

```diff
@@ -167,7 +282,14 @@ func Test_unmarshal_optionality(t *testing.T) {
 					t.Errorf("decodeState.unmarshal() = %s", diff)
 				}
 			default:
-				dst := reflect.New(reflect.TypeOf(tt.in)).Interface()
```

Comment:
Why are can we not treat maps like the other types here? Could we not define dst as the type of the input and unmarshal?

Reply:
if we use `dst := reflect.New(reflect.TypeOf(tt.in)).Interface()` for map,

- it panics in Unmarshal    
>  **panic: assignment to entry in nil map**

-   for `in` of type `map[int8][]byte`, it returns variable of type `*map[int8][]uint8` instead of `*map[int8][]byte`

so need to initialize the map and specify the type of `dst` same as type of `in`.

Reply:
cool cool that makes sense, thank you!